### PR TITLE
adding testing using find and findOne

### PR DESCRIPTION
**Pull request by sibelius**
@alexmingoia 

I've added a test to check if the populate is working properly, I've used the Author Book model, but it is not working, or I using the plugin in a wrong way

I don't know if this is the best way to test callback, it is kind of ugly

Maybe we should migrate to es6, and test using async functions or generators.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -60,4 +76,102 @@ describe('Schema#virtual()', function () {
...
+  //      }
+  //    });
+  //});
+
+  it('should populate correctly', function (done) {
+    mongoose = populateVirtuals(mongoose);
+
+    var bookSchema = new mongoose.Schema({
+      title: { type: String },
+      authorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Author'},
```
Comment (alexmingoia) [inline test/index.js:116]: This plugin changes `.populate()` functionality to populate virtuals instead of objectId fields. This will not work.

Comment (sibelius) [inline test/index.js:116]: So, can I use this plugin to populate a virtual field with documents of another model?

In your example at Readme, you populate the books virtual field of Author with an array of Books that reference the Author, that's right?

What is the right way to implement your example?

Comment (alexmingoia) [inline test/index.js:116]: Sorry, I should have been more clear. Adding a `ref` property to the `authorId` schema will not allow you to: `Book.find(..).populate('authorId')` as it does without using this plugin.

**Comment by alexmingoia** [comment]
There's two reasons this test is failing:
1. You are not waiting for the `.save()` calls to complete before querying for those documents.
2. Mongoose internals have changed since this module was published, and so the virtual [getters/setter](https://github.com/alexmingoia/mongoose-populate-virtuals/blob/master/index.js#L41-L52) are no longer being run. Not sure the best way to fix at the moment.

**Comment by sibelius** [comment]
no problem, I'm using https://github.com/whitecolor/mongoose-fill to achieve the same behavior

**Pull request closed without merging**